presto-orc/src/test/java/com/facebook/presto/orc/TestLongDictionaryProvider.java

```diff
@@ -237,9 +360,52 @@ public NodeId(int node, int sequence)
             this.sequence = sequence;
         }
 
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(node, sequence);
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (o == this) {
+                return true;
+            }
+
+            if (!(o instanceof NodeId)) {
+                return false;
+            }
+
+            NodeId other = (NodeId) o;
+            return other.node == node && other.sequence == sequence;
+        }
+
         public StreamId toDictionaryDataStreamId()
         {
             return new StreamId(node, sequence, DICTIONARY_DATA);
         }
     }
+
+    private static class SharedDictionaryTestCase
```

Comment:
Can we name this class differently ? Case implies a Test case. If there is no better name, we can name it as SharedDictionaryTestInput

Reply:
It is a test case though? It's a struct of inputs and expected states. Test input also sounds good to me.